RHDEVDOCS-6590: Content creation for GitOps 1.18.0 RN (#99180)
openshift/openshift-docs: modules/gitops-release-notes-1-18-0.adoc

[id="gitops-release-notes-1-18-0_{context}"]
= Release notes for {gitops-title} 1.18.0

> {gitops-title} 1.18.0 is available on {OCP} 4.14, 4.15, 4.16, 4.17, 4.18, and 4.19.

Reviewer: ```suggestion
{gitops-title} 1.18.0 is available on {OCP} 4.14, 4.16, 4.17, 4.18, and 4.19.
```

+
link:https://issues.redhat.com/browse/GITOPS-4223[GITOPS-4223]

Support for injecting additional volumes into the Dex container::
With this update, the {gitops-title} Operator supports injecting additional volumes and volume mounts into the Dex container through the Argo CD CR. This enables advanced use cases such as mounting custom certificates, configuration fragments, or identity provider metadata required by Dex.
+
[source,YAML]
----
spec:
  sso:
    provider: dex
    dex:
      volumeMounts:
        - name: custom-cert
          mountPath: /etc/dex/ssl
          readOnly: true
      volumes:
        - name: custom-cert
          secret:
            secretName: dex-tls-secret
----
+
For more information, see "Additional Volume Mounts for Dex Container" in the Additional resources section.
+
link:https://issues.redhat.com/browse/GITOPS-5963[GITOPS-5963]

Control automated synchronization with the `enabled` field::

> With this update, the {gitops-title} Operator introduces a new boolean field named `enabled` under `spec.syncPolicy.automated` in the `Application` CR. The field gives users explicit control over enabling or disabling automated synchronization. The following key improvements are included in this feature:

Reviewer: ```suggestion
With this update, the {gitops-title} Operator introduces a new boolean field named `enabled` under `.spec.syncPolicy.automated` in the `Application` CR. The field gives users explicit control over enabling or disabling automated synchronization. The following key improvements are included in this feature:
```

Author: Fixed.

** A new `Enabled` checkbox in the `Application Create` and `Application Summary` panels.
** Auto-sync can be enabled or disabled independently of `prune` and `selfHeal`.
** Defaults to `true` for backward compatibility.
** When set to `false`, auto-sync is explicitly disabled even if the automated block exists.
+
[source,YAML]
----
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: my-app
  namespace: argocd
spec:
  project: default
  source:
    repoURL: https://github.com/my-org/my-app.git
    targetRevision: HEAD
    path: manifests
  destination:
    server: https://kubernetes.default.svc
    namespace: my-app-namespace
  syncPolicy:
    automated:
      enabled: false # Explicitly disables auto-sync
      prune: true
      selfHeal: true
----
+
[NOTE]
====
If the `enabled` field is not set, automated synchronization defaults to `enabled` to maintain backward compatibility.
====
+
link:https://issues.redhat.com/browse/GITOPS-5430[GITOPS-5430]

Granular control of banner display in the Argo CD UI::

> With this update, the {gitops-title} Operator adds support for the permanent and position properties within the `spec.banner` field in the Argo CD CR. The following options provide more granular control over how banners are displayed in the Argo CD web UI.

Reviewer: ```suggestion
With this update, the {gitops-title} Operator adds support for the permanent and position properties within the `.spec.banner` field in the Argo CD CR. The following options provide more granular control over how banners are displayed in the Argo CD web UI.
```

> * `spec.banner.permanent`: A boolean field, when set to `true`, prevents users from dismissing the banner.

Reviewer: ```suggestion
* `.spec.banner.permanent`: A boolean field, when set to `true`, prevents users from dismissing the banner.
```

> * `spec.banner.position`: A string field that specifies where the banner is displayed. Supported values are `top` and `bottom`.

Reviewer: ```suggestion
* `.spec.banner.position`: A string field that specifies where the banner is displayed. Supported values are `top` and `bottom`.
```

+
[source,YAML]
----
spec:
  banner:
    content: "Maintenance in progress"
    permanent: true
    position: top
----
+
link:https://issues.redhat.com/browse/GITOPS-6679[GITOPS-6679]

Upgraded Kubernetes and Argo CD dependencies::
With this update, the {gitops-title} Operator upgrades the Kubernetes and Argo CD dependencies used in the {gitops-shortname} backend. The newer versions address vulnerability alerts previously flagged by CVE scanners in older dependency versions, ensuring alignment with more recent and secure libraries.
+
link:https://issues.redhat.com/browse/GITOPS-6860[GITOPS-6860]

Default resource exclusions in {gitops-title}::
With this update, the {gitops-title} Operator introduces a default `resource.exclusions` configuration, consistent with upstream Argo CD. By default, the Argo CD Application Controller excludes well-known resources created by controllers and frequently changing or auto-generated resources, which reduces unnecessary load. You can override or customize these exclusions through the Argo CD configuration if needed.
+
link:https://issues.redhat.com/browse/GITOPS-7434[GITOPS-7434]

[id="fixed-issues-1-18-0_{context}"]
== Fixed issues

> Fixed misleading error logs when the `ConsoleLink` resource is disabled::

Reviewer: I am wondering if this needs to be mentioned or not, I was unable to reproduce this behaviour

cc: @nmirasch

Reviewer 2: I’m not completely sure either. My conclusion was that I wasn’t able to modify that env var to disable the console link. From the code, if the env var is set to true, the link is removed and this error should appear even it continues working fine. That said, I didn’t see this log in previous versions, but I don’t think we can conclude that nobody sees it. For me it’s fine to remove this from the release notes.

+
link:https://issues.redhat.com/browse/GITOPS-4223[GITOPS-4223]

Support for injecting additional volumes into the Dex container::
With this update, the {gitops-title} Operator supports injecting additional volumes and volume mounts into the Dex container through the Argo CD CR. This enables advanced use cases such as mounting custom certificates, configuration fragments, or identity provider metadata required by Dex.
+
[source,YAML]
----
spec:
  sso:
    provider: dex
    dex:
      volumeMounts:
        - name: custom-cert
          mountPath: /etc/dex/ssl
          readOnly: true
      volumes:
        - name: custom-cert
          secret:
            secretName: dex-tls-secret
----
+

> For more information, see the upstream documentation for "Additional Volume Mounts for Dex Container".

Author: I have added this link under the Additional Resources section in our release notes. Here is the link for reference: https://99180--ocpdocs-pr.netlify.app/openshift-gitops/latest/release_notes/gitops-release-notes-1-18.html#additional-resources_gitops-release-notes

> For more information, see the upstream documentation for "Additional Volume Mounts for Dex Container".

Reviewer 3: Okay. In that case, can we provide some indication to users about the reference section in this statement? Maybe...

> For more information, see "Additional Volume Mounts for Dex Container" in the Additional resources section below.

> For more information, see the upstream documentation for "Additional Volume Mounts for Dex Container".

Author: Fixed.

kind: Application
metadata:
  name: my-app
  namespace: argocd
spec:
  project: default
  source:
    repoURL: https://github.com/my-org/my-app.git
    targetRevision: HEAD
    path: manifests
  destination:
    server: https://kubernetes.default.svc
    namespace: my-app-namespace
  syncPolicy:
    automated:
      enabled: false # Explicitly disables auto-sync
      prune: true
      selfHeal: true
----
+
[NOTE]
====
If the `enabled` field is not set, automated synchronization defaults to `enabled` to maintain backward compatibility.
====
+
link:https://issues.redhat.com/browse/GITOPS-5430[GITOPS-5430]

Granular control of banner display in the Argo CD UI::
With this update, the {gitops-title} Operator adds support for the permanent and position properties within the `spec.banner` field in the Argo CD CR. The following options provide more granular control over how banners are displayed in the Argo CD web UI.
* `spec.banner.permanent`: A boolean field, when set to `true`, prevents users from dismissing the banner.
* `spec.banner.position`: A string field that specifies where the banner is displayed. Supported values are `top` and `bottom`.
+
[source,YAML]
----
spec:
  banner:

>     text: "Maintenance in progress"

Reviewer 4: Can we replace text with content? So this line would be :
`content: "Maintenance in progress"`

[id="new-features-1-18-0_{context}"]
== New features

Support for defining local users in the Argo CD custom resource::
With this update, you can define local users in the Argo CD custom resource (CR) by using the {gitops-shortname} Operator. The {gitops-shortname} Operator can also automatically generate Argo CD API tokens for these users. For more information about managing and configuring local users in Argo CD, see "Managing local users in Argo CD" in the Additional Resources section.
+
link:https://issues.redhat.com/browse/GITOPS-1874[GITOPS-1874]

Support for tenant-managed namespaces with the `NamespaceManagement` CR::

> With this update, support is provided for the `NamespaceManagement` CR, which enables tenants to allow Argo CD management of their own namespaces without requiring cluster-admin access. Cluster administrators must first enable the feature in the {gitops-title} `Subscription` resource and specify the allowed namespaces in the Argo CD CR. When a tenant creates a `NamespaceManagement` CR in their namespace, the {gitops-title} Operator automatically configures the required `Role` and `RoleBinding` resources and updates Argo CD's managed namespace list. The `NamespaceManagement` CR improves security by reducing the need for elevated permissions, simplifies setup through automatic configuration, and ensures clean resource management by cleaning up configurations when the feature is disabled.

Reviewer 5: The release note is really long and hard to read, the current Flesch–Kincaid grade shows level of 19.85, when the maximum allowed is 12 (and ideal is 9). This needs some rewording/minimalism adjustments to fit the required level.

Author: Fixed in a different PR.

+
link:https://issues.redhat.com/browse/GITOPS-4223[GITOPS-4223]

Support for injecting additional volumes into the Dex container::
With this update, the {gitops-title} Operator supports injecting additional volumes and volume mounts into the Dex container through the Argo CD CR. This enables advanced use cases such as mounting custom certificates, configuration fragments, or identity provider metadata required by Dex.
+
[source,YAML]

> ----

Reviewer 5: There is no introduction to this code block, there should be maybe an introductore sentence, something like: "The following exampl shows..."

Author: Fixed in a different PR.

spec:
  sso:
    provider: dex
    dex:
      volumeMounts:
        - name: custom-cert
          mountPath: /etc/dex/ssl
          readOnly: true
      volumes:
        - name: custom-cert
          secret:
            secretName: dex-tls-secret
----
+
For more information, see "Additional Volume Mounts for Dex Container" in the Additional resources section.
+
link:https://issues.redhat.com/browse/GITOPS-5963[GITOPS-5963]

Control automated synchronization with the `enabled` field::

> With this update, the {gitops-title} Operator introduces a new boolean field named `enabled` under `.spec.syncPolicy.automated` in the `Application` CR. The field gives users explicit control over enabling or disabling automated synchronization. The following key improvements are included in this feature:

Reviewer 5: "you"is more recommended rather then "users", plus removing the word "explicit" for minimalism
```suggestion
With this update, the {gitops-title} Operator introduces a new boolean field named `enabled` under `.spec.syncPolicy.automated` in the `Application` CR. The field gives you control over enabling or disabling automated synchronization. The following key improvements are included in this feature:
```

** A new `Enabled` checkbox in the `Application Create` and `Application Summary` panels.
** Auto-sync can be enabled or disabled independently of `prune` and `selfHeal`.

> ** Defaults to `true` for backward compatibility.

Reviewer 5: The other bulleted items are a full sentence, but this one is incomplete sentence. This should be all the same for consistency, something like:

```suggestion
** The automated synchronization defaults to `true` for backward compatibility.
```

** When set to `false`, auto-sync is explicitly disabled even if the automated block exists.
+
[source,YAML]
----
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: my-app
  namespace: argocd
spec:
  project: default
  source:
    repoURL: https://github.com/my-org/my-app.git
    targetRevision: HEAD
    path: manifests
  destination:
    server: https://kubernetes.default.svc
    namespace: my-app-namespace
  syncPolicy:
    automated:
      enabled: false # Explicitly disables auto-sync
      prune: true

>       selfHeal: true

Reviewer 5: This again should have some introductory statement, such as: "The following example shows disabled automated sync policy:" or something like that. 

Also, only this part:
```
  syncPolicy:
    automated:
      enabled: false # Explicitly disables auto-sync
      prune: true
      selfHeal: true
```
seems relevant, so I would recommand changing th e example to something like this:
```
----
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: my-app
  namespace: argocd
spec:
  syncPolicy:
    automated:
      enabled: false # Explicitly disables auto-sync
      prune: true
      selfHeal: true
  # . . .
----
```

> ----

Reviewer 5: Instead of comments, `# Explicitly disables auto-sync`, this should be outside of the code block (previously we used callouts, now we need to use whatever is in the style guide for examples.

Author: Since I am unaware of how to handle callouts, I am simply removing the comments. I do not feel that it adds enough value.

+
[NOTE]
====
If the `enabled` field is not set, automated synchronization defaults to `enabled` to maintain backward compatibility.

> ====

Reviewer 5: This is already mentioned in the bulleted list. It should be therefore removed.

Author: Fixed in a different PR.

+
link:https://issues.redhat.com/browse/GITOPS-5430[GITOPS-5430]

Granular control of banner display in the Argo CD UI::
With this update, the {gitops-title} Operator adds support for the permanent and position properties within the `.spec.banner` field in the Argo CD CR. The following options provide more granular control over how banners are displayed in the Argo CD web UI.
* `.spec.banner.permanent`: A boolean field, when set to `true`, prevents users from dismissing the banner.
* `.spec.banner.position`: A string field that specifies where the banner is displayed. Supported values are `top` and `bottom`.
+
[source,YAML]

> ----

Reviewer 5: Again, a small introductory sentence should be included for the code block (I know you are probably adding it like this because we no longer can use block headings, but we still need to replace it with something, otherwise users would not have any context to the block codes).

spec:
  banner:
    content: "Maintenance in progress"
    permanent: true
    position: top
----
+
link:https://issues.redhat.com/browse/GITOPS-6679[GITOPS-6679]

Upgraded Kubernetes and Argo CD dependencies::
With this update, the {gitops-title} Operator upgrades the Kubernetes and Argo CD dependencies used in the {gitops-shortname} backend. The newer versions address vulnerability alerts previously flagged by CVE scanners in older dependency versions, ensuring alignment with more recent and secure libraries.
+
link:https://issues.redhat.com/browse/GITOPS-6860[GITOPS-6860]

Default resource exclusions in {gitops-title}::

> With this update, the {gitops-title} Operator introduces a default `resource.exclusions` configuration, consistent with upstream Argo CD. By default, the Argo CD Application Controller excludes well-known resources created by controllers and frequently changing or auto-generated resources, which reduces unnecessary load. You can override or customize these exclusions through the Argo CD configuration if needed.

Author: Fixed.

+
Workaround:
Ensure that your Bitbucket repository name only uses lowercase letters to enable path-based sync triggering.
+
link:https://issues.redhat.com/browse/GITOPS-7835[GITOPS-7835]

[id="deprecated-features-1-18-0_{context}"]
== Deprecated and removed features

Removed support for Keycloak in {gitops-title}::
In {gitops-title} 1.18, the support for Keycloak-based authentication is removed. As an alternative, you can migrate to Dex or configure a self-managed Red Hat Build of Keycloak (RHBK) instance. Dex is the default authentication provider in {gitops-title}. It integrates with the OpenShift OAuth server, enabling users to log in to Argo CD with their existing OpenShift users and groups. For more information, see "Configuring SSO for Argo CD using Dex" in the Additional resources section. If you still want to use Keycloak, you must deploy and manage the "Red Hat Build of Keycloak (RHBK)" instance and manually configure the integration in the Argo CD custom resource. For more information about configuring Argo CD with keycloak, see the upstream documentation for "Keycloak" in the Additional resources section.

> +

Author: Fixed in a different PR.

> +

Author: https://github.com/openshift/openshift-docs/pull/99675
